test/variational/test_exact_state.py

```diff
@@ -122,13 +130,13 @@ def test_qutip_conversion(vstate):
     np.testing.assert_allclose(q_obj.data.todense(), ket.reshape(q_obj.shape))
 
 
-def test_derivatives_agree():
+@pytest.mark.parametrize("machine", machines.values())
```

Comment:
This gives it a more readable name when running under pytest

```suggestion
@pytest.mark.parametrize("machine", [pytest.param(ma, id=name) for name, ma in machines.items()])
```